src/pow.rs

```diff
@@ -270,3 +346,42 @@ fn count_leading_zeros(data: &[u8]) -> u8 {
 
     leading_zeros
 }
+
+/// The faucet will dynamically adjust the difficulty based on the
+/// current balance of the faucet to make it increasingly difficult
+/// to retrieve funds from the faucet. The actual equation for this
+/// is:
+///
+/// ```math
+/// f(x)=(M-m)(1-\log_{q}\frac{x}{b})+m
+/// ```
+///
+/// where:
+///
+/// - `M` is the maximum difficulty
+/// - `m` is the minimum difficulty
+/// - `x` is the current balance in BTC
+/// - `b` is the minimum balance in BTC
+/// - `q` is the amount emitted per request in BTC
+///
+/// # Guarantees
+///
+/// This function guarantees that the difficulty will be between `min_difficulty`
+/// and `max_difficulty` given that the correctness assumptions are met.
+///
+/// # Correctness
+///
+/// For this function correctly output, you must ensure:
+///
+/// - `per_emission` > `Amount::ONE_BTC`, ideally >2 BTC due to the way the curve
+///   functions
+/// - `min_difficulty` <= `max_difficulty`
+/// - `max_difficulty`, `min_difficulty`, `balance`, `min_balance` and `x` all > 0
```

Comment:
There's a redundancy here since `x` represents the current balance.

Reply:
Removed.